testing/internal/e2e/target/target_test.go

```diff
@@ -0,0 +1,280 @@
+	)
+	require.NoError(t, err, string(output.Stderr))
+	var newHostSetResult hostsets.HostSetCreateResult
+	err = json.Unmarshal(output.Stdout, &newHostSetResult)
+	require.NoError(t, err)
+	newHostSet := newHostSetResult.Item
+	t.Cleanup(func() {
+		output, err := e2e.RunCommand("boundary", "host-sets", "delete", "-id", newHostSet.Id)
+		require.NoError(t, err, string(output.Stderr))
+	})
+	t.Logf("Created Host Set: %s", newHostSet.Id)
+
+	// Create a host
+	output, err = e2e.RunCommand(
+		"boundary", "hosts", "create", "static",
+		"-host-catalog-id", newHostCatalog.Id,
+		"-name", c.TargetIp,
+		"-address", c.TargetIp,
+		"-format", "json",
+	)
+	require.NoError(t, err, string(output.Stderr))
+	var newHostResult hosts.HostCreateResult
+	err = json.Unmarshal(output.Stdout, &newHostResult)
+	require.NoError(t, err)
+	newHost := newHostResult.Item
+	t.Cleanup(func() {
+		output, err := e2e.RunCommand("boundary", "hosts", "delete", "-id", newHost.Id)
+		require.NoError(t, err, string(output.Stderr))
+	})
+	t.Logf("Created Host: %s", newHost.Id)
+
+	// Add host to host set
+	output, err = e2e.RunCommand(
+		"boundary", "host-sets", "add-hosts",
+		"-id", newHostSet.Id,
+		"-host", newHost.Id,
+	)
+	require.NoError(t, err, string(output.Stderr))
+
+	// Create a target
+	output, err = e2e.RunCommand(
+		"boundary", "targets", "create", "tcp",
+		"-scope-id", newProject.Id,
+		"-default-port", "22",
+		"-name", "e2e Automated Test Target",
+		"-format", "json",
+	)
+	require.NoError(t, err, string(output.Stderr))
+	var newTargetResult targets.TargetCreateResult
+	err = json.Unmarshal(output.Stdout, &newTargetResult)
+	require.NoError(t, err)
+	newTarget := newTargetResult.Item
+	t.Cleanup(func() {
+		output, err := e2e.RunCommand("boundary", "targets", "delete", "-id", newTarget.Id)
+		require.NoError(t, err, string(output.Stderr))
+	})
+	t.Logf("Created Target: %s", newTarget.Id)
+
+	// Add host set to target
+	output, err = e2e.RunCommand(
+		"boundary", "targets", "add-host-sets",
+		"-id", newTarget.Id,
+		"-host-set", newHostSet.Id,
+	)
+	require.NoError(t, err, string(output.Stderr))
+
+	// Connect to target and print host's IP address
+	output, err = e2e.RunCommand(
+		"boundary", "connect",
+		"-target-id", newTarget.Id,
+		"-exec", "/usr/bin/ssh", "--",
+		"-l", c.TargetSshUser,
+		"-i", c.TargetSshKeyPath,
+		"-o", "UserKnownHostsFile=/dev/null",
+		"-o", "StrictHostKeyChecking=no",
+		"-o", "IdentitiesOnly=yes", // forces the use of the provided key
+		"-p", "{{boundary.port}}", // this is provided by boundary
+		"{{boundary.ip}}",
+		"hostname", "-i",
+	)
+	require.NoError(t, err, string(output.Stderr))
+
+	parts := strings.Fields(string(output.Stdout))
+	hostIp := parts[len(parts)-1]
+	assert.Equal(t, c.TargetIp, hostIp, "SSH session did not return expected output")
+	t.Log("Successfully connected to target")
+}
+
+// TestCreateTargetApi uses the boundary go api to create a number of supporting objects
+// to connect to a target.
```

Comment:
Maybe add a note here that this doesn't attempt to connect to the target because it's complicated to do without the CLI?

Reply:
Personally I'd be curious to see how hard it would be too 😁, but it can wait for another PR, if ever.